lib/datadog/core/runtime/metrics.rb

```diff
@@ -57,15 +57,27 @@ def flush
 
           try_flush do
             if Core::Environment::VMCache.available?
-              gauge(
+              # Only on Ruby < 3.2
+              gauge_if_not_nil(
```

Comment:
> WDYT about an explicit version check? (if or case RUBY_VERSION when ...). That would make the code self-explanatory, and materialise the else case.

I chose not to do the explicit version check so that in the future we wouldn't get bitten by the same issue: let's say Ruby 3.5 removes some of the new metrics, and again we would break if we checked the Ruby version and not the underlying data.

It's true that our comments could become inaccurate in that situation, but since our tests validate the expected results, we would have an indication that there's something we need to fix -- while still not breaking released dd-trace-rb versions.

> WDYT about a feature check? something like if Core::Utils::Features::Ruby.vmcache_global_constant_state?

A feature check per metric would mean more code + more tests to replace the current `nil` check. I wouldn't oppose such a change, but I'm not particularly convinced it's worth the extra complexity either.